tests/beta/tapes/test_tape.py

```diff
@@ -709,3 +709,288 @@ def test_measurement_expansion(self):
+        assert np.allclose(res[1], np.abs(dev.state) ** 2, atol=tol, rtol=0)
+
+    def test_single_mode_sample(self):
+        """Test that there is only one array of values returned
+        for single mode samples"""
+        dev = qml.device("default.qubit", wires=2, shots=10)
+        x = 0.543
+        y = -0.654
+
+        with QuantumTape() as tape:
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            sample(qml.PauliZ(0) @ qml.PauliX(1))
+
+        res = tape.execute(dev)
+        assert res.shape == (1, 10)
+
+    def test_multiple_samples(self):
+        """Test that there is only one array of values returned
+        for multiple samples"""
+        dev = qml.device("default.qubit", wires=2, shots=10)
+        x = 0.543
+        y = -0.654
+
+        with QuantumTape() as tape:
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            sample(qml.PauliZ(0))
+            sample(qml.PauliZ(1))
+
+        res = tape.execute(dev)
+        assert res.shape == (2, 10)
+
+    def test_samples_expval(self):
+        """Test that multiple arrays of values are returned
+        for combinations of samples and statistics"""
+        dev = qml.device("default.qubit", wires=2, shots=10)
+        x = 0.543
+        y = -0.654
+
+        with QuantumTape() as tape:
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            sample(qml.PauliZ(0))
+            expval(qml.PauliZ(1))
+
+        res = tape.execute(dev)
+        assert res[0].shape == (10,)
+        assert isinstance(res[1], float)
+
+    def test_decomposition(self, tol):
+        """Test decomposition onto a devices supported gate set"""
+        dev = qml.device("default.qubit", wires=1)
+
+        with QuantumTape() as tape:
+            qml.U3(0.1, 0.2, 0.3, wires=[0])
+            expval(qml.PauliZ(0))
+
+        tape = tape.expand(stop_at=lambda obj: obj.name in dev.operations)
+        res = tape.execute(dev)
+        assert np.allclose(res, np.cos(0.1), atol=tol, rtol=0)
+
+
+class TestCVExecution:
+    """Tests for CV tape execution"""
+
+    def test_single_output_value(self, tol):
+        """Tests correct execution and output shape for a CV tape
+        with a single expval output"""
+        dev = qml.device("default.gaussian", wires=2)
+        x = 0.543
+        y = -0.654
+
+        with QuantumTape() as tape:
+            qml.Displacement(x, 0, wires=[0])
+            qml.Squeezing(y, 0, wires=[1])
+            qml.Beamsplitter(np.pi / 4, 0, wires=[0, 1])
+            expval(qml.NumberOperator(0))
+
+        assert tape.output_dim == 1
+
+        res = tape.execute(dev)
+        assert res.shape == (1,)
+
+    def test_multiple_output_values(self, tol):
+        """Tests correct output shape and evaluation for a tape
+        with multiple expval outputs"""
```

Comment:
test covers a slightly different case than the docstring

Reply:
```suggestion
        with multiple measurement types
```